test/functional/api/cas/git.py

```diff
@@ -36,29 +52,17 @@ def get_release_tags():
 
 
 def checkout_cas_version(cas_version):
-    from api.cas.version import CasVersion
-    if isinstance(cas_version, CasVersion):
-        output = TestRun.executor.run(
-            f"cd {TestRun.usr.working_dir} && "
-            f"git rev-parse {cas_version}")
-        if output.exit_code != 0:
-            raise CmdException(f"Failed to resolve {cas_version} tag to commit hash", output)
-        TestRun.LOGGER.info(f"Resolved {cas_version} as commit {output.stdout}")
-        cas_version = output.stdout
-
-    _checkout_cas_commit(cas_version)
-
-
-def _checkout_cas_commit(commit_hash):
+    commit_hash = get_commit_hash(cas_version)
     TestRun.LOGGER.info(f"Checkout CAS to {commit_hash}")
+
     output = TestRun.executor.run(
         f"cd {TestRun.usr.working_dir} && "
         f"git checkout --force {commit_hash}")
     if output.exit_code != 0:
-        raise CmdException(f"Failed to checkout to CAS {commit_hash}", output)
+        raise CmdException(f"Failed to checkout to {commit_hash}", output)
 
     output = TestRun.executor.run(
         f"cd {TestRun.usr.working_dir} && "
-        f"git submodule update")
+        f"git submodule update --force")
```

Comment:
This `--force` makes me a bit uncomfortable with `LocalExecutor`... I'm not sure I want tests to go and overwrite my repo state - I'd rather it errored out and let me know it failed

Reply:
All the executor operations are performed on the repo which resides in the 'working_dir', which is '/tmp/open-cas-linux'. Repo which resides there is constantly a subject to various actions - it is removed, changed, rsynced, etc. by many steps performed by test-framework.
So forcing the submodule update there is not the issue. It is not changing your local repo, just a copy of it.
Besides, if you look 6 lines above, there is a similar '--force' flag when checking out to a different revision (and this would be more dangerous, as it discards your unsaved changes). But again - this is only a working copy of the repo. :)

Reply:
Oh, sorry, didn't catch that, carry on!